wordpress-mobile/WordPress-iOS
# Restore autosave revision dialog

### Overview
Fixes #12141

On the post list screen, posts with an autosave revision are now marked with a **new status label**. When tapped, instead of just loading the post as before, users are now also given the option to **load the autosaved revision**. Loading the autosave revision works in both the Gutenberg and Aztec editors.

### Demo
<img src="https://user-images.githubusercontent.com/1898325/69395356-7042ce00-0cbd-11ea-80dd-a810843d7238.gif" width=350/>

### Testing Steps
1. On WordPress.com, create a new post or open an existing post for editing
2. Set a new arbitrary post title, content, and excerpt
3. Publish the post
4. Re-open the post and make an edit – to the **title, content and/or excerpt** – then click the preview button to trigger an autosave in the background
5. On the iOS app, under the My Sites tab, navigate to the list of posts containing the newly created post
6. Observe that there is now a status label – _"You've made unsaved changes to this post"_ –indicating that an autosave revision is available
7. Tap the post and observe a dialog giving an option to load the post directly or load the autosaved revision
8. Tapping the option to load autosave revision ("From another device") should load the changes made in Step 4 into the editor (if there is an autosaved excerpt, it appears in the editor menu ➔ Post Settings ➔ Excerpt)
9. Tapping "Update" should save the changes and if the user exits the editor, the changes should be shown on the post list screen
10. Likewise, the option to load the post directly ("From this device") should load the post without the changes from Step 4
11. If the autosave option is chosen and the user makes edits and then closes the editor without saving, the post list should still show the status label from Step 5 and subsequently tapping on the post should again show the dialog

### Notes
- The above steps must work for,
  - both the Gutenberg and Aztec editors, and switching between editors must work as expected
  - both published and scheduled posts
- Posts with local changes must **_not_** offer load autosave functionality
- An autosave can be created on the same device the post is later loaded on - so the dialog labels "From this device" and "From another device" don't always seem intuitive
  - Similarly, if the autosave is created on a device (e.g. changes are previewed), then the user exits the editor and discards the changes, the option to load the autosaved (discarded) changes is still available

Update release notes:
